src/utils/getValidSubjectIconName.ts

```diff
@@ -0,0 +1,20 @@
+import { OakIconName, isValidIconName } from "@oaknational/oak-components";
+
+import errorReporter from "@/common-lib/error-reporter";
+import OakError from "@/errors/OakError";
+
+export const getValidSubjectIconName = (subject: string): OakIconName => {
+  const subjectIconName = `subject-${subject}`;
+  if (!isValidIconName(subjectIconName)) {
+    const reportError = errorReporter("generateSubjectIconName");
+    reportError(
+      new OakError({
+        code: "oak-components/invalid-icon-name",
+        meta: { iconName: subjectIconName },
+      }),
+    );
+    // fallback icon
+    return "error";
```

Comment:
very good point! @Monika-Koziol do you have any preference on which icon we should use as a fallback (in case we can't get the subject specific icon for any reason)?